Sources/Query.swift

```diff
@@ -8,159 +8,507 @@
+    case circle(center: CLLocationCoordinate2D, radius: Double)
+}
+
+/// All the possible MIME types that are supported by Contentful. \
+//  Developer note: Cases in String backed Swift enums have a raw value equal to the case name.
+//  i.e. MimetypeGroup.attachement.rawValue = "attachment"
+public enum MimetypeGroup: String {
+    case attachment
+    case plaintext
+    case image
+    case audio
+    case video
+    case richtext
+    case presentation
+    case spreadsheet
+    case pdfdocument
+    case archive
+    case code
+    case markup
+}
+
+/** 
+ Property-value query operations used for matching patterns on either "sys" or "fields" properties of `Asset`s and `Entry`s.
+ Each operation specifies a property name on the left-hand side, with a value to match on the right.
+ For instance, using the doesNotEqual operation in an a concrete Query like:
+ 
+    ```
+    Query(where:"fields.name", .doesNotEqual("Happy Cat"))
+    ```
+ 
+ would append the following to the http URL:
+
+    ```
+    "fields.name[ne]=Happy%20Cat"
+    ```
+ 
+ Refer to the documentation for the various Query classes for more information.
+*/
 public enum QueryOperation {
+
     case equals(String)
     case doesNotEqual(String)
     case hasAll([String])
     case includes([String])
     case excludes([String])
     case exists(Bool)
 
-    internal var operation: String {
-        switch self {
-        case .equals:
-            return ""
+    /// Full text search on a field.
+    case matches(String)
+
+    /// MARK: Ranges
+    case isLessThan(QueryableRange)
+    case isLessThanOrEqualTo(QueryableRange)
+    case isGreaterThan(QueryableRange)
+    case isGreaterThanOrEqualTo(QueryableRange)
 
-        case .doesNotEqual:
-            return "[ne]"
+    /// Proximity searches.
+    case isNear(CLLocationCoordinate2D)
+    case isWithin(Bounds)
 
-        case .hasAll:
-            return "[all]"
+    fileprivate var string: String {
+        switch self {
+        case .equals:                                       return ""
+        case .doesNotEqual:                                 return "[ne]"
+        case .hasAll:                                       return "[all]"
+        case .includes:                                     return "[in]"
+        case .excludes:                                     return "[nin]"
+        case .exists:                                       return "[exists]"
+        case .matches:                                      return "[match]"
+
+        case .isLessThan:                                   return "[lt]"
+        case .isLessThanOrEqualTo:                          return "[lte]"
+        case .isGreaterThan:                                return "[gt]"
+        case .isGreaterThanOrEqualTo:                       return "[gte]"
+
+        case .isNear:                                       return "[near]"
+        case .isWithin:                                     return "[within]"
```

Comment:
Do you have the [query](https://www.contentful.com/developers/docs/references/content-delivery-api/#/reference/search-parameters/full-text-search) somehow useable or just through the dictionary?